### ISSUE TITLE
Multiple Crawlers

### ISSUE THREAD

**Issue by schemen** [boramalper/magnetico#106]
Hi!

Quick question. Is it possible to run multiple crawlers with a shared network file system extending the same database?

Scenario: I got it running via Docker on a Synology system which is slow, I got another server, could they extend each other?

Cheers!

**Comment by ad-m** [comment]
SQLite used in magnetico* is not optimized for network applications. It would be valuable to add support to database in client-server architecture. See https://sqlite.org/whentouse.html .

**Comment by Glandos** [comment]
Maybe magneticod should stay as-is, using one database file per-instance.
However, magneticow can be modified to query multiple databases. This would lead to some duplicates, but code for magneticod persistence would stay simple.

**Comment by ad-m** [comment]
I think that abstraction and two storage implementations would not complicate the code too much

**Comment by Glandos** [comment]
Yes, of course, abstraction layer for accessing SQL database is quite easy nowadays.
But I was also thinking that it could be a good feature for magneticow to have multiple sources.

**Comment by ad-m** [comment]
Why do you want to have multiple data sources? In magneticow or in mangeticod? What are the obstacles to the centralization of the database to keep simplicity?

**Comment by Glandos** [comment]
I'm talking about multiple data sources in magneticow. The use case is when you can have multiple magneticod running on different hosts far away from each other. magneticod needs a fast connection to its database to be able to find duplicates quickly (as from today), whereas magneticow can takes more time to answer a user request.

**Comment by ad-m** [comment]
@Glandos , replication is a solution.

**Comment by Glandos** [comment]
Fortunately, there often is multiple solutions for a single problem in software engineering ;)

**Comment by ad-m** [comment]
Yes, but I think it is worth choosing solutions that will make magnetico* a simple solution that is usable without the title of IT professor. Delegation of issues to external optional components helps this. 

I wonder if relational databases are optimal for us anyway.

**Comment by skobkin** [comment]
> Maybe magneticod should stay as-is, using one database file per-instance.
> However, magneticow can be modified to query multiple databases

So you'll get data duplication across all databases. What are your initial goal? Speeding up the crawling process or data replication?
